### ISSUE TITLE
Pep8-speaks to Actions linter

### ISSUE THREAD

**Issue by alexdaniel654** [UKRIN-MAPS/ukat#140]
When ukat went public pep8speaks seems to have broken. There are some other, probably better, options out there now Actions have taken off a bit more. I would suggest replacing pep8speaks with [this](https://github.com/marketplace/actions/lintly-flake8).

**Comment by JSousa-UoL** [comment]
I agree with this action. pep8speaks was the possible solution at the time due to `ukat` being a private repository. Now that it's public, we should implement the Github action you're suggesting.

**Comment by JSousa-UoL** [comment]
I did some investigation about this recently and have been testing different pep8 linting github actions projects (including the one suggested in your first comment). Feel free to have a look at the closed draft PR #180 to check all the attempts (a bit messy though).

None of them worked because I was making a pull request from my fork JSousa-UoL. Didn't test this, but from what I read online, they would have worked if the PR from another branch. Then it would be possible to have a PR with PEP8 comments from the github-actions bot. When it comes to PEP8 comments in PRs from external forks, there are 2 options. One is pep8speaks, which is what we have in ukat and it's a managed Github App. The other option would be to use the `pull_request_target` event in a CI workflow. I wanted to try this in the PR #180 but I didn't because it seems there are serious security risks associated. See this [link](https://securitylab.github.com/research/github-actions-preventing-pwn-requests/) for details.

In conclusion, I suggest we close this issue because we already have the right and secure linter implementation in pep8speaks.
